lib/Dialect/ArithExt/IR/ArithExtOps.td

```diff
@@ -14,6 +14,35 @@ class ArithExt_Op<string mnemonic, list<Trait> traits = [Pure]> :
   let cppNamespace = "::mlir::heir::arith_ext";
 }
 
+class ArithExt_BinaryOp<string mnemonic, list<Trait> traits = []> :
+    ArithExt_Op<mnemonic, traits # [SameOperandsAndResultType, Pure, ElementwiseMappable]>,
+    Arguments<(ins SignlessIntegerLike:$lhs, SignlessIntegerLike:$rhs, I64Attr:$modulus)>,
```

Comment:
Why not use an arbitrary width IntegerAttr here instead of a fixed bit width modulus?

Reply:
Indeed! I just need #715 (which adds lowerings for `arith_ext.barrett_reduce` and `arith_ext.subifge`) and this PR to be merged before #731 can do that across all ops & lowerings

Reply:
Got it